src/app/user/settings/page.tsx

```diff
@@ -1,231 +1,337 @@
+      setError(err?.message ?? "Password update failed.");
     }
   }
 
   async function handleDelete() {
-    resetAlerts();
-    if (!auth.currentUser) return;
-
     if (!confirm("Are you sure you want to delete your account?")) return;
+    resetAlerts();
 
     try {
-      if (currentEmail && currentPassword) {
-        const credential = EmailAuthProvider.credential(
-          currentEmail,
-          currentPassword
-        );
-        await reauthenticateWithCredential(auth.currentUser, credential);
-      }
-      await deleteUser(auth.currentUser);
-      setSuccess("Account deleted successfully.");
-      logout();
+      if (!confirmPassword.trim())
+        throw new Error("Enter your password first.");
+      await reauthWithPassword(currentEmail, confirmPassword);
+      await removeAccount();
+      await logout();
     } catch (err: any) {
-      setError(err.message ?? "Delete failed.");
+      setError(err?.message ?? "Delete failed.");
     }
   }
 
   return (
-    <div className="min-h-dvh flex items-center justify-center bg-gray-400 dark:bg-gray-700/70">
-      <Card size="sm" className="min-w-[300px]">
+    <div className="min-h-dvh flex items-center justify-center bg-gray-400 dark:bg-gray-700/70 px-4">
+      <Card size="md" className="w-full max-w-xl dark:bg-gray-300">
         <Title
           label="Settings"
           align="center"
-          className="text-slate-800 dark:text-slate-100 mb-3"
+          className="text-slate-100 dark:text-slate-800 mb-3"
         />
 
-        <div className="flex gap-2 justify-center mb-3">
-          {["email", "password", "delete"].map((t) => (
-            <button
-              key={t}
-              type="button"
-              onClick={() => {
-                resetAlerts();
-                setTab(t as Tab);
-              }}
-              className={`px-4 py-2 text-sm font-medium rounded-lg border transition-all duration-200  ${
-                tab === t
-                  ? "bg-gray-400 text-black dark:bg-slate-700 dark:text-white shadow-sm"
-                  : "border-slate-300 text-slate-700 hover:bg-gray-200 dark:text-slate-300 dark:hover:bg-slate-800"
-              }`}
-            >
-              {t === "email"
-                ? "Update Email"
-                : t === "password"
-                ? "Update Password"
-                : "Delete Account"}
-            </button>
-          ))}
-        </div>
+        <div
+          className="rounded-lg border border-slate-300 dark:border-slate-700 p-3 
+                           bg-gray-100 dark:bg-slate-900"
+        >
+          <p className="text-center text-sm mb-4">
+            Current email: <strong>{currentEmail}</strong>{" "}
+          </p>
+
+          <div className="flex gap-2 justify-center mb-8 mt-3">
+            {(["email", "password", "delete"] as Tab[]).map((t) => (
+              <button
+                key={t}
+                type="button"
+                onClick={() => {
```

Comment:
It's a bad practice to add multiline function directly in the onClick events. It's vary hard to read/maintain code like this.